docs/src/explanations/advanced-data-types.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: docs
+title:  "Advanced Data Types"
+section: "chisel3"
+---
+
+# Advanced Data Types
```

Comment:
I don't love this title for this doc. "Advanced Data Types" implies thinking about more expressive and powerful types (think `Records` compared to `Bundles` or learning about Higher-Kinded Types in Scala). This is more of "Chisel vs. Scala Types") because fundamentally, the Chisel types are types in Chisel's type system, it's just that Chisel's type system is a Scala runtime thing.